# links

https://gitter.im/UZH-BIO392/BIO392-2022-course-chat

https://drive.switch.ch/index.php/s/iCTAmYzVZEMSWYP

https://compbiozurich.org/courses/UZH-BIO392/

https://progenetix.org

https://github.com/compbiozurich/UZH-BIO392


# day 3 

- UCSC genome browser https://genome-euro.ucsc.edu/cgi-bin/hgTracks?db=hg38&lastVirtModeType=default&lastVirtModeExtraState=&virtModeType=default&virtMode=0&nonVirtPosition=&position=chrX%3A15560138%2D15602945&hgsid=290662285_Fb1M2aMP5aCk8N4Foq8zPwovXDPC
  - genome vs. conservation, exons, .. -> find spots which likely contain lots of information.
  - data representation is "boring plain text"
  - 3bio bo/human genome
  - variant data store logic: chrom, start, end (of variant, duplication, ..?)
    - discussed storage formats: only reference genoma, FASTA and one other (FASTQ?) sequence based
  - sequencing: degree of uncertainty for each nucleotide
  - naming of reference genomes is a mess
  - "if youre doing science and youre using a mouse: this is not reproducible"
  
## terminal, UNIX

 

















